unikernel.ml

```diff
@@ -1328,6 +1328,102 @@ struct
             Middleware.http_response reqd ~title:"Error"
               ~data:"Couldn't find fields" `Bad_request)
 
+  let download_volume json_dict albatross reqd (user : User_model.user) =
+    match
+      Utils.Json.(get "block_name" json_dict, get "compression_level" json_dict)
+    with
+    | Some (`String block_name), Some (`Int compression_level) -> (
+        Albatross.query albatross ~domain:user.name ~name:block_name
+          (`Block_cmd (`Block_dump compression_level))
+        >>= function
+        | Error msg ->
+            Logs.err (fun m -> m "Error querying albatross: %s" msg);
+            Middleware.http_response reqd ~title:"Error"
+              ~data:("Error querying albatross: " ^ msg)
+              `Internal_server_error
+        | Ok (_hdr, res) -> (
+            match Albatross_json.res res with
+            | Ok res ->
+                let file_content = Yojson.Basic.to_string res in
+                let filename = block_name ^ "_dump" in
+                let disposition = "attachment; filename=\"" ^ filename ^ "\"" in
+                reply reqd ~content_type:"application/octet-stream"
+                  ~header_list:[ ("Content-Disposition", disposition) ]
+                  file_content `OK
+                |> Lwt.return
+            | Error (`String res) ->
+                Middleware.http_response reqd ~title:"Error"
+                  ~data:(Yojson.Basic.to_string (`String res))
+                  `Internal_server_error))
+    | _ ->
+        Middleware.http_response reqd ~title:"Error"
+          ~data:"Couldn't find block name in json" `Bad_request
+
+  let upload_to_volume albatross reqd (user : User_model.user) =
+    read_multipart_data reqd >>= fun result ->
```

Comment:
similar to unikernel_deploy, we should (at one day in our live) have a streaming interface here as well, instead of requiring the entire data fitting into memory.

the downside (and we'll likely need to revise the protocol) is that in TLS we can embed only 16MB of certificates, this means that we can't write/set multi-gigabyte block devices. for now, this is a shortcoming -- I opened an issue at the albatross repository https://github.com/robur-coop/albatross/issues/201